src/xmpp_callbacks.ml

```diff
@@ -34,10 +34,14 @@ type user_data = {
 
 let validate_utf8 txt =
   let txt = String.map (fun c -> match c with
-      | '\x00' .. '\x08' |'\x0b'|'\x0c'| '\x0e' .. '\x1f'
-       (* filter <0x20 except if it's tab (x09) or newline (x0a/x0d) *)
-        -> '?'
-      | c -> c
+       (* filter <0x20 except if it's tab (x09) or newline (x0a) *)
+      | '\x09' | '\x0a'
+          -> c
+      | '\x0d' (* CR, turn it into a LF / newline *)
```

Comment:
This will turn a CR LF into two newlines which feels wrong. There are also a few other newline characters in Unicode. You may want to have a look at uutf's [newline normalization](http://erratique.ch/software/uutf/doc/Uutf#TYPEnln) rather than pile up the hacks.